crates/bevy_render/src/texture/image_texture_conversion.rs

```diff
@@ -1,152 +1,158 @@
 
-            data = raw_data.as_slice().as_bytes().to_owned();
-        }
+                let raw_data = i.into_raw();
 
-        image::DynamicImage::ImageRgb16(image) => {
-            width = image.width();
-            height = image.height();
-            format = TextureFormat::Rgba16Uint;
-
-            let mut local_data =
-                Vec::with_capacity(width as usize * height as usize * format.pixel_size());
-
-            for pixel in image.into_raw().chunks_exact(3) {
-                // TODO unsafe_get in release builds?
-                let r = pixel[0];
-                let g = pixel[1];
-                let b = pixel[2];
-                let a = u16::max_value();
-
-                local_data.extend_from_slice(&r.to_ne_bytes());
-                local_data.extend_from_slice(&g.to_ne_bytes());
-                local_data.extend_from_slice(&b.to_ne_bytes());
-                local_data.extend_from_slice(&a.to_ne_bytes());
+                data = raw_data.as_slice().as_bytes().to_owned();
             }
 
-            data = local_data;
-        }
-        image::DynamicImage::ImageRgba16(i) => {
-            width = i.width();
-            height = i.height();
-            format = TextureFormat::Rgba16Uint;
+            image::DynamicImage::ImageRgb16(image) => {
+                width = image.width();
+                height = image.height();
+                format = TextureFormat::Rgba16Uint;
+
+                let mut local_data =
+                    Vec::with_capacity(width as usize * height as usize * format.pixel_size());
 
-            let raw_data = i.into_raw();
+                for pixel in image.into_raw().chunks_exact(3) {
+                    // TODO unsafe_get in release builds?
+                    let r = pixel[0];
+                    let g = pixel[1];
+                    let b = pixel[2];
+                    let a = u16::max_value();
 
-            data = raw_data.as_slice().as_bytes().to_owned();
+                    local_data.extend_from_slice(&r.to_ne_bytes());
+                    local_data.extend_from_slice(&g.to_ne_bytes());
+                    local_data.extend_from_slice(&b.to_ne_bytes());
+                    local_data.extend_from_slice(&a.to_ne_bytes());
+                }
+
+                data = local_data;
+            }
+            image::DynamicImage::ImageRgba16(i) => {
+                width = i.width();
+                height = i.height();
+                format = TextureFormat::Rgba16Uint;
+
+                let raw_data = i.into_raw();
+
+                data = raw_data.as_slice().as_bytes().to_owned();
+            }
         }
+
+        Texture::new(
+            Extent3d::new(width, height, 1),
+            TextureDimension::D2,
+            data,
+            format,
+        )
     }
+}
 
-    Texture::new(
-        Extent3d::new(width, height, 1),
-        TextureDimension::D2,
-        data,
-        format,
-    )
+pub enum TextureConversionError {
```

Comment:
```suggestion
#[derive(Clone, Copy, Debug)] // also maybe Eq/PartialEq
pub enum TextureConversionError {
```

Also is it worth implementing `std::error::Error` for this?